seed/seedwriter/seed16.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package seedwriter
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/snap/naming"
+)
+
+type policy16 struct {
+	model *asserts.Model
+	opts  *Options
+
+	needsCore   []string
+	needsCore16 []string
+}
+
+func (pol *policy16) systemSnap() *asserts.ModelSnap {
+	if pol.model.Classic() {
+		// no predefined system snap, infer later
+		return nil
+	}
+	snapName := "core"
+	snapType := "core"
+	if pol.model.Base() != "" {
+		snapName = "snapd"
+		snapType = "snapd"
+	}
+	// TODO: set SnapID too
+	return &asserts.ModelSnap{
+		Name:           snapName,
+		SnapType:       snapType,
+		Modes:          []string{"run"},
```

Comment:
Do we need modes in policy16?

Reply:
that's what asserts code landed does for ModelSnaps, it seems more consistent to set and just know that Core 16/18 atm have onyl run